sdk/nodejs/output.ts

```diff
@@ -389,49 +398,28 @@ export function isSecretOutput<T>(o: Output<T>): Promise<boolean> {
     return Output.isInstance(o.isSecret) ? Promise.resolve(false) : o.isSecret;
 }
 
-/**
- * [output] takes any Input value and converts it into an Output, deeply unwrapping nested Input
- * values as necessary.
- *
- * The expected way to use this function is like so:
- *
- * ```ts
- *      var transformed = pulumi.output(someVal).apply(unwrapped => {
- *          // Do whatever you want now.  'unwrapped' will contain no outputs/promises inside
- *          // here, so you can easily do whatever sort of transformation is most convenient.
- *      });
- *
- *      // the result can be passed to another Resource.  The dependency information will be
- *      // properly maintained.
- *      var someResource = new SomeResource(name, { data: transformed ... });
- * ```
- */
-export function output<T>(val: Input<T>): Output<Unwrap<T>>;
-export function output<T>(val: Input<T> | undefined): Output<Unwrap<T | undefined>>;
-export function output<T>(val: Input<T | undefined>): Output<Unwrap<T | undefined>> {
+interface ValueInfo {
+    value: any;
+    isOutput?: boolean;
+}
+
+function outputRec(val: any): ValueInfo {
```

Comment:
Needs a docstring and should be marked `internal`

Reply:
These aren't exported--do they need to be marked `internal`?

Reply:
Whoops, you're right. Disregard